native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -27,7 +27,9 @@ class ViewController: UIViewController, ApplicationContractView {
     @IBAction func departureButton(_ sender: Any) {
         let departureStation: String = stationData[departurePicker.selectedRow(inComponent: 0)]
         let arrivalStation: String = stationData[arrivalPicker.selectedRow(inComponent: 0)]
-        presenter.onButtonTapped(departureStation: departureStation, arrivalStation: arrivalStation, view: self)
+        presenter.setDepartureStation(departureStation: departureStation)
```

Comment:
Think this was just a temporary fix to get things running with the updated `onButtonTapped` method - we have another branch where there is a listener on the picker so the station is selected then rather than when the button is pressed